shared/AppInsightsCommon/src/StorageHelperFuncs.ts

```diff
@@ -30,56 +32,156 @@ function _getLocalStorageObject(): Storage {
  * @returns {Storage} Returns storage object verified that it is usable
  */
 function _getVerifiedStorageObject(storageType: StorageType): Storage {
+    let result = null;
+    const storageTypeName = storageType === StorageType.LocalStorage ? "localStorage" : "sessionStorage";
+    
     try {
-        if (isNullOrUndefined(getGlobal())) {
-            return null;
+        // Default to false - assume storage is not available
+        let canAccessStorage = false;
+        const gbl = getGlobal();
+        
+        // Only proceed if we have a global object
+        if (!isNullOrUndefined(gbl)) {
+            // Try the safest method first (property descriptor)
+            try {
+                const descriptor = objGetOwnPropertyDescriptor(gbl, storageTypeName);
+                if (descriptor && descriptor.get) {
+                    canAccessStorage = true;
+                }
+            } catch (e) {
+                // If descriptor check fails, try direct access
+                // This will be caught by the outer try-catch if it fails
+                canAccessStorage = !!gbl[storageTypeName];
+            }
         }
-        let uid = (new Date).toString();
-        let storage: Storage = getGlobalInst(storageType === StorageType.LocalStorage ? "localStorage" : "sessionStorage");
-        let name:string = _storagePrefix + uid;
-        storage.setItem(name, uid);
-        let fail = storage.getItem(name) !== uid;
-        storage.removeItem(name);
-        if (!fail) {
-            return storage;
+        
+        // If we determined storage might be accessible, verify it works
+        if (canAccessStorage) {
+            try {
+                const uid = (new Date).toString();
+                const storage: Storage = getGlobalInst(storageTypeName);
+                const name = _storagePrefix + uid;
+                
+                storage.setItem(name, uid);
+                const success = storage.getItem(name) === uid;
+                storage.removeItem(name);
+                
+                if (success) {
+                    // Create a wrapped storage object that protects write operations
+                    const originalStorage = storage;
+                    const wrappedStorage = {
+                        // Read operations - don't reset cache on error
+                        getItem: function(key: string): string {
+                            try {
```

Comment:
Create a helper to avoid the multiple sets of boilerplate code.
ie. a function that returns the implementation and as part of the main function it gets past the function name.